text/trace/0226-sampling-random-traceids.md

```diff
@@ -0,0 +1,205 @@
+printf format specifier.  Values in the range [0x1p-56, 0x1p+56] are
+valid.  Values in the range [0x1p-56, 1] are interpreted as a sampling
+probability, while values in the range [1, 0x1p+56] are intepreted as
+an adjusted count.  Adjusted count values must be integers, while
+sampling probability values can be arbitrary floating point values.
+
+Whether to encode sampling probabilty or adjusted count is a choice.
+In both cases, the interpreted value translates into an exact
+threshold, which determines the exact inclusion probability.  From the
+exact inclusion probability, we can determine the adjusted count to
+use in a span-to-metrics pipeline.  When the t-value is _stated_ as an
+adjusted count (as opposed to a sampling probabilty), implementations
+can use the integer value in a span-to-metrics pipeline.  Otherwise,
+implementations should use an adjusted count of 1 divided by the
+sampling probability.
+
+## Where to store t-value in a Span and/or Log Record
+
+Although prepared as a solution for tail sampling, the t-value
+encoding scheme could also be used to convey Logs sampling.  While
+tail sampling does not require the use of trace state, which is
+associated with context propagation, it makes a natural place to store
+t-value because it should be interpreted along with p-value, which
+resides in the trace state.  However, if spans store t-value in trace
+state, it is not clear how to convey logs sampling.
+
+Here are ways to address this:
+
+1. Store t-value in a new dedicated field in the Span or Log Record
+   (as a string).  (Author's preference.)
+2. Store t-value as a Span or Log Record attribute (as a string).
+   This may cause confusion because the attribute, which was not
+   applied by a user, can change long the collection path even though
+   the data has not changed.
+3. Store t-value as an optional floating point field in the Span or
+   Log Record.  An optional field is required because we need a
+   meaningful way to represent zero probability, for cases where spans
+   are exporter due to a non-probabilistic decision.
+4. Create a new field in both Spans and Log Records as a dedicated
+   field for storing t-values.
+   
+The benefit of using TraceState is that it is an extensible field,
+made for multiple vendors to place arbitrary contents.  It is not
+clear whether use of tracestate to record collection-time decisions is
+appropriate, or whether it is only meant for in-band context
+propagation.  If this use-case is acceptable, the name Trace State
+would become a legacy; in this case, a more signal-neutral name for
+the field could be developed (e.g., "Collection State")
+
+### 90% sampling 
+
+The following header
+
+```
+tracestate: ot=t:0.9
+```
+
+### 1-in-3 sampling
+
+The following header
+
+```
+tracestate: ot=t:3
+```
+
+corresponds with 1-in-3 sampling.
+
+### 25% head sampling, 1-in-10 tail sampling
+
+The following header
+
+```
+tracestate: ot=p:2;t:10
```

Comment:
Just for clarity, I'd recommend that we use 3 here and a 1-in-8 / 80 example below, so it's more obvious it's not a mistake.

Reply:
I now think it would be best to deprecate the p-value, r-value mechanism, so I removed this example that combined them.